tests/transforms/test_metric_tensor.py

```diff
@@ -579,6 +579,71 @@ def layer3_diag(params):
         G_expected = block_diag(G1, G3, G2)
         assert qml.math.allclose(G, G_expected, atol=tol, rtol=0)
 
+    @pytest.mark.parametrize("interface", ["autograd", "jax", "tf", "torch"])
+    def test_argnum_metric_tensor(self, tol, interface):
+        """Test that argnum successfully reduces the number of tapes and gives
+        the desired outcome."""
+        dev = qml.device("default.qubit", wires=3)
+
+        def circuit(weights):
+            qml.RX(weights[0], wires=0)
+            qml.RY(weights[1], wires=0)
+            qml.CNOT(wires=[0, 1])
+            qml.RZ(weights[2], wires=1)
+            qml.RZ(weights[3], wires=0)
+
+        weights = np.array([0.1, 0.2, 0.3, 0.5], requires_grad=True)
+
+        with qml.tape.QuantumTape() as tape:
+            circuit(weights)
+
+        tapes, proc_fn = qml.metric_tensor(tape)
+        res = qml.execute(tapes, dev, None)
+        mt = proc_fn(res)
+        zeros = qml.math.zeros_like(mt)
+
+        tapes, proc_fn = qml.metric_tensor(tape, argnum=(0, 1, 3))
+        res = qml.execute(tapes, dev, None, interface=interface)
+        mt013 = proc_fn(res)
+
+        assert len(tapes) == 6
+        assert mt.shape == mt013.shape
+        assert qml.math.allclose(mt[:2, :2], mt013[:2, :2], atol=tol, rtol=0)
+        assert qml.math.allclose(mt[3, 3], mt013[3, 3], atol=tol, rtol=0)
+        assert qml.math.allclose(zeros[2, :], mt013[2, :], atol=tol, rtol=0)

```

Comment:
Just as a super minor side comment, this works as well (as we already checked the shape..):
```suggestion
        assert qml.math.allclose(0, mt013[2, :], atol=tol, rtol=0)
```
:)

Reply:
Ah, thanks!